Controllers/CustomersController.cs

```diff
@@ -38,52 +38,99 @@ public async Task<JsonResult> GetData()
                 : Json(new { Status = "Failure" });
         }
 
-        public async Task<IActionResult> CustomerLogin()
+        public async Task<IActionResult> CustomerLogin(bool? clickLinktoChange)
         {
+            var customerCount = 0;
             //RETURN LIST OF CUSTOMERS THAT A MEMBER OF
             //IF ONLY ONE, TAKE THEM HOME.  IF MULTIPLE, PROVIDE LIST AND USER SELECTS WHICH CUSTOMER TO LOGIN TO
-            var currentUser = await _recitopiaDbContext.AppUsers
+            if (User.Identity.Name != null)
+            {
+
+                var currentUser = await _recitopiaDbContext.AppUsers
                 .SingleAsync(m => m.UserName == User.Identity.Name);
 
-            var customerIds = await _recitopiaDbContext.Customer_Users
-                .Where(cu => cu.Id == currentUser.Id)
-                .Select(cu => cu.Customer_Id)
-                .ToListAsync();
+                var customerIds = await _recitopiaDbContext.Customer_Users
+                    .Where(cu => cu.Id == currentUser.Id)
+                    .Select(cu => cu.Customer_Id)
+                    .ToListAsync();
+            
+                customerCount = customerIds.Count();
 
-            if (customerIds.Count() > 1)
-            {
-                //Provide selection view
+                if (customerCount > 1 && clickLinktoChange != true)
+                {
 
-                //create list and populate with Customer name and Id
-                var custList = new List<List<string>>();
+                    //Look to see if company id and name saved prior.  If so, bypass selection page and take to home
+                    var checkLastLoginCompanyInfo = _recitopiaDbContext.AppUsers.Where(m => m.Id == currentUser.Id).First();
```

Comment:
If more than one user can't have the same id, you should use the proper methods to show your intent and requirements. 

`_recitopiaDbContext.AppUser.Single(x => x.Id == currentUser.Id)` 

`Single() `will throw an exception if there is not exactly one record.
`SingleOrDefault()` will throw an exception if there is more than one record but return default (null for reference types) if there is none.
`First() `will return the first record in the collection
`FirstOrDefault()` will return the first record or default (null for reference types) if there is none